### PR TITLE
[ui] Code location: Asset page

### PR DESCRIPTION
**Pull request by hellendag**
## Summary & Motivation

Add an asset list page to the new code location definitions UI. These are grouped by asset group, and are collapsible like in other areas of the app. I've reused the existing grouping/flattening code from the current workspace asset list, and added some improved sorting as well.

Screenshot 2024-08-20 at 10 26 17

Screenshot 2024-08-20 at 10 26 24

## How I Tested These Changes

Storybook example.

### CODE REVIEW

**Comment by hellendag** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/23755?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

* **#23755** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/23755?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* **#23738** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/23738?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* `master`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @hellendag and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: js_modules/dagster-ui/packages/ui-core/src/assets/AssetLink.tsx [hunk 1]
```diff
@@ -44,3 +34,17 @@ export const AssetLink = (props: {
...
+export const buildAssetPathString = (path: string[], isGroup?: boolean) => {
+  return (
+    path
+      .reduce((accum, elem, ii) => {
+        if (ii > 0) {
+          accum.push(' / ');
+        }
+        accum.push(elem);
+        return accum;
+      }, [] as string[])
```
Comment (bengotow) [inline js_modules/dagster-ui/packages/ui-core/src/assets/AssetLink.tsx:47]: Would `path.join(' / ')` work? We also have `displayNameForAssetKey` which might do this now,

I think this code was very very old :-)

**Review by bengotow — APPROVED** [review]
This looks great!

**Comment by hellendag** [comment]
### Merge activity

* **Aug 21, 11:10 AM CDT**: @hellendag started a stack merge that includes this pull request via [Graphite](https://app.graphite.dev/github/pr/dagster-io/dagster/23755).

**Pull request merged**